docs/development/journalist_api.rst

```diff
@@ -0,0 +1,414 @@
+              "filesystem_id": "LBICF2DPGI3AMD74HIYXQ7QKQB3MCCKNMQ6Q4VPOL3OYWMIDMYDO6A37KOJCZNT3GVOEMCDHECM4S4OAXGGMZ452SD454A6EADXN3ZI=",
+              "flagged": false,
+              "interaction_count": 2,
+              "journalist_designation": "clairvoyant burdock",
+              "last_updated": "Fri, 29 Jun 2018 19:11:30 GMT",
+              "number_of_documents": 0,
+              "number_of_messages": 2,
+              "public_key": "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\nmQINBFs2hGQBEACnIkg5HQpABY/Rpmf8GhN96xqrEBABtK60FgomzdydGUlCip29\nPLzlMVFaAuGNJyo2S2izJr8n8TXmQYAQMP+OGdc+33In047NSCgF3ZGblUkexYKy\n/q8/Jr8YdLDeonJpYG0uQLnA2AA8FJucadkZCc30MPh+g7iPoKsmoRmr32GEpttS\n0XIfzjBhrc3uX1pEH8g9NP1CCHjbkLV1uY/Zo7svwPfbeEicXuK2TEl7ovlx8WYt\nz52sBwfsory2Eyy9D21IUKVBU1tWWeQeTAJrovg+auBZTwSV2+sYM7nE1zjWDDtA\nUSabvtP6O8dDO+vAMxmO80JxYONGfrS0XO5FSATpiApwsxS7o9ZSri3N+vLDQez4\npEQ0dkGa1NgTaUSVDzh+XIFWugd00wWg/rC6d3pZSjZXOA+p7BVUMsAfCLUZMxgz\n7JiqgZhM6TQ/RfReeSYDeUVT5ioImfDsOB79GArt+uvbesLxwLzoAcL6RWtqdK6k\nEcy277g7V5zsASJE6FAaYxS9dkqg9Zc+oSzlNtF7G0Kg3HIjZDwLoG+NzI7f4cMv\nXVka+GSHlWsElgE1My2HryC/SzqeVBbpg0vM8QaIMxiDrnLtjrD28L9Hi/5ab7Rq\nRF43lWWXQeEbKQ6nxLhQrVsM3E1xYx+JJLTBEJbNUo+TwTN7vfhAOpNJ4wARAQAB\ntHxBdXRvZ2VuZXJhdGVkIEtleSA8TEJJQ0YyRFBHSTNBTUQ3NEhJWVhRN1FLUUIz\nTUNDS05NUTZRNFZQT0wzT1lXTUlETVlETzZBMzdLT0pDWk5UM0dWT0VNQ0RIRUNN\nNFM0T0FYR0dNWjQ1MlNENDU0QTZFQURYTjNaST0+iQI/BBMBCgApBQJbNoRkAhsv\nBQkB4M5cBwsJCAcDAgEGFQgCCQoLBBYCAwECHgECF4AACgkQZmOkQ/49FwH1Yw/+\nIHhA2QpvDyThSwWthuh3ytdOJ9VveLO1jtBmDkuZtU/wpMgyVdCMusCOszXePSt5\n3neAcVOYFUBgKQTCmGAOXY8hOMNwHcdl13/ehiAwdj+BvE1OIBdLplCwW41F4esv\nvPvxBQW47oeRNt+u15keNXpWQBjFbB894yWQFlIn6sfEgvB9E53M2UHHn3NUzjKy\nIhC+ItMAodvEPpj34PAVPRxYk3TQkzsA/q9J48nAhY04x7lhSBp8M+jU07iGR2hB\nsewE/cwO5CVew7T7R5b1tl8iGIPmPeb4+zLc2xXy/oBAFRqI0BVdMskhtpmmvUzr\nScKN6GjX9a4TpOhxm3msyeKt5dnc3uOp3e7CBsDnYOTavDHeKvrkKZKukuvAXGt5\ne3RAITcvuOLVdswchwiex3HXq/rrvRHIglBaE56ZKo8XOm9+zBrcZzLjTmY1DChB\nhZGBX2p5tcZEN2h7n04BzFuPGNRB/PJa3A0qc3/aX3sJ8gGTovEt93Yzz6XyM70m\nBoo04NPwFv6JhEIm/qsbGTSFJO5NPONpaZ/54AKMldbIaq56eXz2si3Ltrl1pPIv\nqdmuW0VxMMt0l3xPZe3sBzNfp6MnWGjVYHfTIsXHbHgZWJKiMrhW9o2UjsmNlXUJ\n0asrUWe/LIDPk/5mB42CX1O6lwEkuo7uGoCa2F+8efs=\n=RC5t\n-----END PGP PUBLIC KEY BLOCK-----\n",
+              "remove_star_url": "/api/v1/sources/LBICF2DPGI3AMD74HIYXQ7QKQB3MCCKNMQ6Q4VPOL3OYWMIDMYDO6A37KOJCZNT3GVOEMCDHECM4S4OAXGGMZ452SD454A6EADXN3ZI%3D/remove_star/",
+              "reply_url": "/api/v1/sources/LBICF2DPGI3AMD74HIYXQ7QKQB3MCCKNMQ6Q4VPOL3OYWMIDMYDO6A37KOJCZNT3GVOEMCDHECM4S4OAXGGMZ452SD454A6EADXN3ZI%3D/reply/",
+              "source_id": 2,
+              "submissions_url": "/api/v1/sources/LBICF2DPGI3AMD74HIYXQ7QKQB3MCCKNMQ6Q4VPOL3OYWMIDMYDO6A37KOJCZNT3GVOEMCDHECM4S4OAXGGMZ452SD454A6EADXN3ZI%3D/submissions/",
+              "url": "/api/v1/sources/LBICF2DPGI3AMD74HIYXQ7QKQB3MCCKNMQ6Q4VPOL3OYWMIDMYDO6A37KOJCZNT3GVOEMCDHECM4S4OAXGGMZ452SD454A6EADXN3ZI%3D/"
+          }
+      ]
+  }
+
+Individual Source ``[/sources/<filesystem_id>]``
```

Comment:
Thought about this again. We return the `source_id` in the response. Why not use that?

Reply:
Right now our choices of slug are:

 - `source_id` - slightly bad because it's the primary key in the database i.e. it provides information in URLs about how many sources are in the system, not a huge deal now, but potentially an issue in the future if e.g. HTTPS on the journalist interface is ever configured. Reviewers of the initial API spec/implementation took issue with this so I stopped using it in this PR. 
 - `filesystem_id` - this is consistent with what we do on the journalist interface application, but in addition to the concern you note about the trailing `/`, this is also the hash of the source's passphrase. 

What I mentioned elsewhere was that we should be using a slugified `journalist_designation` (similar to the `journalist_filename` attribute). I think the thing to do here is simply to just implement this `journalist_designation_slug` column, with the database migration that entails (at this time the corresponding change should be made over on the journalist interface). 

I'll wait for a bit before making this change so that observers can issue a "nay" if they strongly disagree.

Reply:
Yeah I see your point. I think if we're adding a slugified field, why not just a UUID? What does adding a human readable string buy us?